src/sagemaker_xgboost_container/data_utils.py

```diff
@@ -528,33 +528,50 @@ def _get_pipe_mode_files_path(data_path: Union[List[str], str]) -> List[str]:
     return files_path
 
 
+def _make_symlinks_for_files_under_a_folder(dest_path: str, data_path: str):
+    if (not os.path.exists(dest_path)) or (not os.path.exists(data_path)):
+        raise exc.AlgorithmError("Unable to create symlinks as {data_path} or {dest_path} doesn't exist ")
+
+    logging.info("Making smlinks from folder {} to folder {}".format(data_path, dest_path))
+
+    if os.path.isfile(data_path):
+        _make_symlink(data_path, dest_path, os.path.basename(data_path))
+        return
+
+    else:
+        for item in os.scandir(data_path):
+            if item.is_file():
+                _make_symlink(item.path, dest_path, item.name)
+            elif item.is_dir():
+                _make_symlinks_for_files_under_a_folder(dest_path, item.path)
```

Comment:
Do we need to worry about recursion depth here?

Reply:
emmm, I was thinking about this too. I will add a depth limit of 5 here.